webpack/assets/javascripts/react_app/components/HostDetails/index.js

```diff
@@ -132,25 +114,34 @@ const HostDetails = ({ match, location: { hash } }) => {
           </Text>
           <br />
         </div>
-        <Tabs
-          style={{
-            width: window.innerWidth - (isNavCollapsed ? 95 : 220),
-          }}
-          activeKey={activeTab}
-          onSelect={handleTabClick}
-        >
-          {tabs &&
-            tabs.map(tab => (
-              <Tab eventKey={tab} title={tab}>
-                <Slot
-                  response={response}
-                  status={status}
-                  id="host-details-page-tabs"
-                  fillID={tab}
-                />
-              </Tab>
-            ))}
-        </Tabs>
+        <HashRouter>
+          <>
+            <Tabs
+              style={{
+                width: window.innerWidth - (isNavCollapsed ? 95 : 220),
+              }}
+              activeKey={hash.slice(2)}
+            >
+              {tabs &&
+                tabs.map(tab => (
+                  <Tab eventKey={tab} title={tab} href={`#${tab}`} />
+                ))}
+            </Tabs>
+            <Switch>
+              {tabs?.map(tab => (
+                <Route path={`/${tab}`}>
+                  <Slot
+                    response={response}
+                    status={status}
+                    id="host-details-page-tabs"
+                    fillID={tab}
+                    router={history}
```

Comment:
@xprazak2, this addition is for creating links outside `hashRouter` scope

```js
<a onClick={() => router.push(`/url`)> Link </a> } // results with "/url" not with "#/url" 
```

Alternatively, we can use the primary browser router instead and manage hashing navigation ourselves.